probe_src/probe_frontend/src/metadata.rs

```diff
@@ -0,0 +1,19 @@
+use machine_info::{Machine, SystemInfo};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Metadata {
+    entry_pid: libc::pid_t,
+    arch: &'static str,
+    system: SystemInfo,
+}
+
+impl Metadata {
```

Comment:
This is cool. I'm wondering if I need to turn it into an op. Maybe `InitRootProc` or `InitProv`. That way, the Python consumers have to only parse 1 thing (ops) not 2 (ops and metadat). Let me know your thoughts.

Reply:
Yeah maybe, I chucked it in last minute as a first take at trying to include non-captured systems state like kernel version and architecture, I'm totally fine with seeing this change. For what it's worth, this struct is bigger than `Op`, while I think some of the info in `SystemInfo` isn't necessary, I wouldn't want to increase the size of the `Op` struct for an op that there should only be one of. 

If you think this should be covered by libprobe itself, I'll take it out and then I or someone else can open a PR for that at a later date.

Reply:
Having separated the recording and transcription processes, metadata as an op seems the way to go; I haven't deleted it yet, but the metadata code is currently unused, either way, I'd say that's out of scope for this PR since it'll probably involve changes to libprobe.